visual-dotnet/SauceLabs.Visual/VisualClient.cs

```diff
@@ -220,22 +225,39 @@ private async Task FinishBuild(VisualBuild build)
         /// <param name="name">the name of the screenshot</param>
         /// <param name="options">the configuration for the screenshot capture and comparison</param>
         /// <returns></returns>
-        public async Task<string> VisualCheck(string name, VisualCheckOptions? options = null)
+        public Task<string> VisualCheck(string name, VisualCheckOptions? options = null,
+            [CallerMemberName] string callerMemberName = "")
+        {
+            options ??= new VisualCheckOptions();
+            if (!string.IsNullOrEmpty(callerMemberName) && (string.IsNullOrEmpty(options.SuiteName) || string.IsNullOrEmpty(options.TestName)))
```

Comment:
small nit:
maybe we can keep the business logic here clean and expose it to a method or util?